testdata/nil-len.go

```diff
@@ -35,4 +35,6 @@ func fn() {
 
 	if s != nil && len(s) == 0 { // nil check is not redundant here
 	}
+	if s != nil && len(s) != 1 { // nil check is not redundant here
```

Comment:
this test case isn't right - this nil check _is_ redundant. it would be not redundant if the left hand side was not a constant, e.g. `if s != nil && len(s) != someOtherValue {`.